src/renderer/cursor_renderer/mod.rs

```diff
@@ -102,17 +108,15 @@ impl Corner {
         font_dimensions: Point,
         destination: Point,
         dt: f32,
+        mode: u64,
     ) -> bool {
         // Update destination if needed
         let mut immediate_movement = false;
 
         if destination != self.previous_destination {
-            let travel_distance = destination - self.previous_destination;
-            let chars_travel_x = travel_distance.x / font_dimensions.x;
-
-            if travel_distance.y == 0.0 && (chars_travel_x - 1.0).abs() < 0.1 {
-                // We're moving one character to the right. Make movement immediate to avoid lag
-                // while typing
+            // 2 for insert mode as it's the only mode where
+            // the cursor might appear to lag behind
+            if !settings.animate_in_insert_mode && mode == 2 {
```

Comment:
I'm nervous about using the 2 constant here. Ideally we should search for the mode which has a name of Insert (or whatever the mode name is that gets reported). This index could change in the future breaking things.

What do you think?

Reply:
totally agree, i wanted to have the current mode as an editor property but this seemed like easy cheat till we have proper mode handling in the editor type. perhaps i should rework that then